src/Network/OAuth/OAuth2/Compat.hs

```diff
@@ -48,14 +68,14 @@ fetchAccessToken
     :: Manager
     -> OAuth2
     -> ExchangeToken
-    -> IO (OAuth2Result Errors OAuth2Token)
+    -> IO (OAuth2Result Error OAuth2Token)
```

Comment:
Why can't we avoid this diff?

```hs

#if MIN_VERSION_hoauth2(2,7,0)
type Errors = TokenRequestError
#else
type Errors = OAuth2Error OAuth2.Errors
#endif

type OAuth2Result err a = Either err a

-- ...

   -> IO (OAuth2Result Errors OAuth2Token)
```

Sorry if this is re-hashing trodden ground. I'm not blocking on it.